packages/web/src/components/track/DownloadRow.tsx

```diff
@@ -0,0 +1,55 @@
+import { useCallback } from 'react'
+
+import { CommonState, ID, cacheTracksSelectors } from '@audius/common'
+import { Flex, IconReceive, PlainButton, Text } from '@audius/harmony'
+import { shallowEqual, useSelector } from 'react-redux'
+
+import { Icon } from 'components/Icon'
+
+const { getTrack } = cacheTracksSelectors
+
+type DownloadRowProps = {
+  trackId: ID
+  onDownload: (trackId: ID, category?: string, parentTrackId?: ID) => void
+}
+
+export const DownloadRow = ({ trackId, onDownload }: DownloadRowProps) => {
+  const track = useSelector(
+    (state: CommonState) => getTrack(state, { id: trackId }),
+    shallowEqual
+  )
+
+  const handleClick = useCallback(() => {
+    if (track !== undefined && track?.track_id !== undefined) {
+      onDownload(track.track_id, track.stem_of?.category, trackId)
```

Comment:
kinda weird that this takes `track.track_id` and also `trackId`

Reply:
also, somewhere we should pass in `is_original` to the DN `/v1/download` endpoint based on if user is downloading original v mp3. check out that route's query params and how they work

Reply:
will follow up in separate PR for that.